File: generate-icon.js
const fs = require('fs');
const { createCanvas } = require('canvas');

const width = 128;
const height = 128;
const canvas = createCanvas(width, height);
const ctx = canvas.getContext('2d');

// Background
ctx.fillStyle = '#2D2D30';
ctx.roundRect(0, 0, 128, 128, 16);
ctx.fill();

// Terminal window
ctx.strokeStyle = '#007ACC';
ctx.lineWidth = 2;
ctx.fillStyle = '#1E1E1E';
ctx.roundRect(16, 24, 96, 64, 4);
ctx.fill();
ctx.stroke();

// Terminal prompt
ctx.fillStyle = '#4EC9B0';
ctx.font = 'bold 16px monospace';
ctx.fillText('$', 24, 48);
ctx.fillRect(36, 38, 32, 3);

// Speaker icon
ctx.fillStyle = '#FFD700';
ctx.beginPath();
ctx.moveTo(72, 44);
ctx.lineTo(72, 60);
ctx.lineTo(80, 60);
ctx.lineTo(88, 68);
ctx.lineTo(88, 36);
ctx.lineTo(80, 44);
ctx.closePath();
ctx.fill();

// Sound waves
ctx.strokeStyle = '#FFD700';
ctx.lineWidth = 3;
ctx.lineCap = 'round';
ctx.beginPath();
ctx.moveTo(88, 44);
ctx.quadraticCurveTo(96, 44, 96, 52);
ctx.quadraticCurveTo(96, 60, 88, 60);
ctx.stroke();

ctx.lineWidth = 2.5;
ctx.globalAlpha = 0.7;
ctx.beginPath();
ctx.moveTo(92, 40);
ctx.quadraticCurveTo(104, 40, 104, 52);
ctx.quadraticCurveTo(104, 64, 92, 64);
ctx.stroke();

ctx.lineWidth = 2;
ctx.globalAlpha = 0.5;
ctx.beginPath();
ctx.moveTo(96, 36);
ctx.quadraticCurveTo(112, 36, 112, 52);
ctx.quadraticCurveTo(112, 68, 96, 68);
ctx.stroke();

// Status indicators
ctx.globalAlpha = 1;
ctx.fillStyle = '#4EC9B0';
ctx.beginPath();
ctx.arc(28, 100, 6, 0, Math.PI * 2);
ctx.fill();

ctx.fillStyle = '#F48771';
ctx.beginPath();
ctx.arc(48, 100, 6, 0, Math.PI * 2);
ctx.fill();

ctx.fillStyle = '#FFD700';
ctx.beginPath();
ctx.arc(68, 100, 6, 0, Math.PI * 2);
ctx.fill();

// Save
const buffer = canvas.toBuffer('image/png');
fs.writeFileSync('icon.png', buffer);
console.log('Icon created: icon.png');

File: generate-icon.py
from PIL import Image, ImageDraw, ImageFont
import os

# Create image
img = Image.new('RGBA', (128, 128), (45, 45, 48, 255))
draw = ImageDraw.Draw(img)

# Terminal window
draw.rounded_rectangle([(16, 24), (112, 88)], radius=4, fill=(30, 30, 30, 255), outline=(0, 122, 204, 255), width=2)

# Terminal prompt line
draw.rectangle([(36, 38), (68, 41)], fill=(78, 201, 176, 255))

# Speaker icon (simplified)
speaker_points = [(72, 44), (72, 60), (80, 60), (88, 68), (88, 36), (80, 44)]
draw.polygon(speaker_points, fill=(255, 215, 0, 255))

# Sound waves (arcs)
draw.arc([(88, 44), (96, 60)], 270, 90, fill=(255, 215, 0, 255), width=3)
draw.arc([(92, 40), (104, 64)], 270, 90, fill=(255, 215, 0, 180), width=2)
draw.arc([(96, 36), (112, 68)], 270, 90, fill=(255, 215, 0, 128), width=2)

# Status indicators (circles)
draw.ellipse([(22, 94), (34, 106)], fill=(78, 201, 176, 255))  # Success - green
draw.ellipse([(42, 94), (54, 106)], fill=(244, 135, 113, 255))  # Failure - red
draw.ellipse([(62, 94), (74, 106)], fill=(255, 215, 0, 255))    # Warning - yellow

# Terminal prompt symbol
try:
    font = ImageFont.truetype("consola.ttf", 16)
except:
    try:
        font = ImageFont.truetype("Courier New.ttf", 16)
    except:
        font = ImageFont.load_default()

draw.text((24, 38), "$", fill=(78, 201, 176, 255), font=font)

# Save
img.save('icon.png')
print("Icon created successfully: icon.png")
